zerver/lib/user_groups.py

```diff
@@ -11,8 +11,14 @@ def access_user_group_by_id(user_group_id: int, user_profile: UserProfile) -> Us
     try:
         user_group = UserGroup.objects.get(id=user_group_id, realm=user_profile.realm)
         group_member_ids = get_user_group_members(user_group)
-        msg = _("Only group members and organization administrators can administer this group.")
-        if not user_profile.is_realm_admin and user_profile.id not in group_member_ids:
+        msg = _(
+            "Only group members, organization administrators and moderators can administer this group."
+        )
+        if (
+            not user_profile.is_realm_admin
+            and not user_profile.is_moderator
+            and user_profile.id not in group_member_ids
+        ):
             raise JsonableError(msg)
```

Comment:
I think it would be better to make this just `raise JsonableError(_("Insufficient permission"))`, like we have in the decorator.  Once the strings for explaining the configuration in error message get really complicated like this, it's usually better to stop trying to have these ultra-specific error messages that explain the full configuration.

(And I think we make pretty clear in the UI who has permissions, so it shouldn't be hard for the user to debug).

If in some future world, we want to have ultra-specific error messages again, we'll likely want to do that in some way that's common to the `has_permission` framework.